.cloudbuild/scripts/cmd/unit-tests/main.go

```diff
@@ -106,6 +110,30 @@ func run() error {
 		return trace.Wrap(err)
 	}
 
+	// If a github deploy key location was supplied...
+	var deployKey []byte
+	if args.githubKeySrc != "" {
+		// fetch the deployment key from the GCB secret manager
+		log.Infof("Fetching deploy key from %s", args.githubKeySrc)
+		deployKey, err = secrets.Fetch(context.Background(), args.githubKeySrc)
+		if err != nil {
+			return trace.Wrap(err, "failed fetching deploy key")
+		}
+	}
+
+	log.Info("Configuring git")
+	gitCfg, err := git.Configure(args.workspace, deployKey)
+	if err != nil {
+		return trace.Wrap(err, "failed configuring git")
+	}
+	defer gitCfg.Close()
+
+	log.Info("Unshallowing repository")
+	err = gitCfg.Do("fetch", "-v", "--unshallow")
```

Comment:
Now that we have more control, could we perhaps only fetch the target branch instead? (same for `integration-tests`)

Reply:
Yes, perhaps. I'm going to vote for "not now", but I agree it's a good idea.

Reply:
Doing some tests locally, I can't seem to find a good way to tell git to actually fetch an additional branch "properly", the best I could come up with is

```go
	err = gitCfg.Do("fetch", "-v", "--depth=1", "origin", "FETCH_HEAD")
```

and then we compare to FETCH_HEAD later on, but that's obviously a terrible solution.